src/app/calendar/_components/CalendarSection.tsx

```diff
@@ -0,0 +1,49 @@
+"use client";
+import { Calendar } from "@/components/Calendar";
+import { useState } from "react";
+import { CalendarTimeManager } from "./CalendarTimeManager";
+import { CalendarDaySection } from "./CalendarDaySection";
+
+enum ReservationType {
+  Lecture = "Lecture",
+  Consultation = "Consultation",
+  Exam = "Exam",
+}
+
+type Reservation = {
+  title: string;
+  startTime: string;
+  endTime: string;
+  date: Date;
+};
+
+export function CalendarSection() {
+  const [selectedDate, setSelectedDate] = useState<Date | undefined>(
+    new Date()
+  );
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <section className="flex">
+      <div className="grid grid-cols-2 maxSM:grid-cols-1 maxSM:grid-rows-2 rounded-lg">
+        <Calendar
+          mode="single"
+          selected={selectedDate}
+          onDayClick={(day: Date) => {
```

Comment:
you shouldn't need to type it implicitly as Date
```suggestion
          onDayClick={(day) => {
```